sdk/storage/Azure.Storage.DataMovement/tests/Shared/StartTransferDirectoryDownloadTestBase.cs

```diff
@@ -0,0 +1,708 @@
+            CancellationToken cancellationToken = default)
+        {
+            await SetupSourceDirectoryAsync(sourceContainer, sourcePrefix, itemSizes, cancellationToken);
+
+            using DisposingLocalDirectory disposingLocalDirectory = DisposingLocalDirectory.GetTestDirectory();
+
+            // Set transfer options
+            options ??= new DataTransferOptions();
+            TestEventsRaised testEventsRaised = new TestEventsRaised(options);
+
+            transferManagerOptions ??= new TransferManagerOptions()
+            {
+                ErrorHandling = DataTransferErrorMode.ContinueOnFailure
+            };
+
+            StorageResourceContainer sourceResource = GetStorageResourceContainer(sourceContainer, sourcePrefix);
+            LocalFilesStorageResourceProvider localProvider = new();
+            StorageResourceContainer destinationResource = localProvider.FromDirectory(disposingLocalDirectory.DirectoryPath);
+
+            await new TransferValidator().TransferAndVerifyAsync(
+                sourceResource,
+                destinationResource,
+                GetSourceLister(sourceContainer, sourcePrefix),
+                TransferValidator.GetLocalFileLister(disposingLocalDirectory.DirectoryPath),
+                itemSizes.Count,
+                options,
+                cancellationToken);
+        }
+
+        [Test]
+        [LiveOnly] // https://github.com/Azure/azure-sdk-for-net/issues/33082
+        [TestCase(0, 10)]
+        [TestCase(100, 10)]
+        [TestCase(Constants.KB, 10)]
+        public async Task DownloadDirectoryAsync_Small(int size, int waitInSec)
+        {
+            // Arrange
+            await using IDisposingContainer<TContainerClient> test = await GetDisposingContainerAsync();
+            string sourceDirectoryName = "foo";
+
+            List<string> itemNames = new()
+            {
+                string.Join("/", sourceDirectoryName, GetNewObjectName()),
+                string.Join("/", sourceDirectoryName, GetNewObjectName()),
+                string.Join("/", sourceDirectoryName, "bar", GetNewObjectName()),
+                string.Join("/", sourceDirectoryName, "bar", "pik", GetNewObjectName()),
+            };
+
+            CancellationTokenSource cts = new();
+            cts.CancelAfter(TimeSpan.FromSeconds(waitInSec));
+            await DownloadDirectoryAndVerifyAsync(
+                test.Container,
+                sourceDirectoryName,
+                itemNames.Select(name => (name, size)).ToList(),
+                cancellationToken: cts.Token).ConfigureAwait(false);
+        }
+
+        [Ignore("These tests currently take 40+ mins for little additional coverage")]
+        [Test]
+        [LiveOnly]
+        [TestCase(4 * Constants.MB, 20)]
+        [TestCase(257 * Constants.MB, 500)]
+        [TestCase(400 * Constants.MB, 200)]
+        [TestCase(Constants.GB, 500)]
+        public async Task DownloadDirectoryAsync_Large(int size, int waitInSec)
```

Comment:
I know this is a pattern we've followed everywhere but I wonder if it's worth having some slightly larger transfers actually run Live, or maybe in Weekly (if that's a thing in .NET). Maybe not the 500 GiB one but maybe up to 1 GiB or something would be fine to run Live only.
Just a thought for the future, not really on this PR directly.

Reply:
Ideally these will go into our stress test framework rather than here. But we need that up and running first.